src/web3/KeychainAccountManager.ts

```diff
@@ -194,43 +188,16 @@ export class KeychainSigner implements Signer {
     await storePrivateKey(privateKey, this.account, passphrase)
   }
 
-  async signTransaction(
-    addToV: number,
-    encodedTx: RLPEncodedTx
-  ): Promise<{ v: number; r: Buffer; s: Buffer }> {
-    Logger.info(`${TAG}@signTransaction`, `Signing transaction:`, encodedTx.transaction)
-    const { gasPrice } = encodedTx.transaction
-    const gasPriceBN = new BigNumber((gasPrice || 0).toString())
-    if (gasPriceBN.isNaN() || gasPriceBN.isLessThanOrEqualTo(0)) {
-      // Make sure we don't sign and send transactions with 0 gas price
-      // This resulted in those TXs being stuck in the txpool for nodes running geth < v1.5.0
-      throw new Error(`Preventing sign tx with 'gasPrice' set to '${gasPrice}'`)
-    }
-
-    return this.localSigner.signTransaction(addToV, encodedTx)
-  }
-
-  async signPersonalMessage(data: string): Promise<{ v: number; r: Buffer; s: Buffer }> {
-    Logger.info(`${TAG}@signPersonalMessage`, `Signing ${data}`)
-    return this.localSigner.signPersonalMessage(data)
-  }
-
-  async signTypedData(typedData: EIP712TypedData): Promise<{ v: number; r: Buffer; s: Buffer }> {
-    Logger.info(`${TAG}@signTypedData`, `Signing typed DATA:`, { address: this.account, typedData })
-    return this.localSigner.signTypedData(typedData)
-  }
-
-  getNativeKey() {
-    return this.account.address
-  }
-
   async unlock(passphrase: string, duration: number): Promise<boolean> {
     const privateKey = await getStoredPrivateKey(this.account, passphrase)
     if (!privateKey) {
       return false
     }
-
-    this.unlockedLocalSigner = new LocalSigner(privateKey)
+    this.localContractKitSigner = new ContractKitSigner(privateKey, this.account)
```

Comment:
Interesting -- it looks like you're using the pattern we currently use for unlocking, which is to just create a totally new instance of the signer, overwrite the old one, and presumably let it get GC'd. Naively I would have expected this higher-level unlock to call an unlock method on the lower-level signers, and the signers themselves would be instantiated once and exist forever. Is there a reason why we do it this way instead?

Reply:
> Naively I would have expected this higher-level unlock to call an unlock method on the lower-level signers, and the signers themselves would be instantiated once and exist forever

Naively that is what I thought would happen too, and what I tried to do first. The notion of locking/unlocking is not something that wallets (or signers) support. Its purely an application construct. As to why it is implemented in this way by overwriting with totally new instances, I'm not sure. @jeanregisser do you know why?